### DISCUSSION TITLE
Performance of Direct3D 11 is worse than OpenGL

### DISCUSSION THREAD

**Discussion by cihe13375** [ppy/osu#23027] [Q&A]
Hi! I have just tried out the latest (2023.326.1) version of lazer. The change log says that the DirectX has the best performance on Windows thus is the default. However, I noticed that the Direct3D 11 renderer performs significantly worse than OpenGL/OpenGL(legacy) on my computer:

OpenGL/OpenGL(legacy): 40\~50 fps in song select, 90\~100 fps in a typical map (watching autoplay, with no storyboard or movie)
Direct3D 11 (default): \~30 fps in song select, 70\~80 fps in the same map
stable (for reference): \~110 fps in song select, \~130 fps in the same map
All runs performed at fullscreen mode (borderless doesn't make things better).

When using Direct3D 11 I notice the entire system being less responsive when running osu (even in the background), with the CPU running at \~1.0 GHz, with power consumption of \~10W (being capped by PL1, according to ThrottleStop; indeed I observed performance improvement after lifting PL1 manually). The task manager shows osu! occupying \~90% of GPU.
After switching to OpenGL the issue is gone: the CPU can boost up to 2.5~3 GHz and the system runs way smoother, with a bit less power consumption (8\~9W). The task manager shows osu! occupying \~10% of GPU.

Platform: Surface Pro 6 (i5-8250U, 8GB RAM), win10 21H2(19044.2728), Intel UHD 620 driver 30.0.101.1340, 2736x1824 screen resolution.

**Comment by peppy** [comment]
Thanks for the feedback! We'll continue to optimise things and let you know if we need further information from you. For now, feel free to switch to the renderer which works the best for you.

**Comment by kireikosasha** [comment]
you have shit videodriver with bad directx support xdd
